third_party/move/move-vm/runtime/src/loader/type_loader.rs

```diff
@@ -16,50 +16,97 @@ pub fn intern_type(
     tok: &SignatureToken,
     struct_name_table: &[StructNameIndex],
 ) -> PartialVMResult<Type> {
-    let list = |toks: &[SignatureToken]| {
-        toks.iter()
-            .map(|tok| intern_type(module, tok, struct_name_table))
-            .collect::<PartialVMResult<Vec<_>>>()
-    };
+    Ok(intern_type_impl(module, tok, struct_name_table)?.0)
+}
+
+/// Converts a stream of signature tokens into the in memory type representation used by the
+/// MoveVM. Also returns true if all tokens are non-generic (contain 0 type parameters).
+pub fn intern_types(
+    module: BinaryIndexedView,
+    toks: &[SignatureToken],
+    struct_name_table: &[StructNameIndex],
+) -> PartialVMResult<(Vec<Type>, bool)> {
+    intern_types_impl(module, toks, struct_name_table)
+}
+
+fn intern_types_impl(
+    module: BinaryIndexedView,
+    toks: &[SignatureToken],
+    struct_name_table: &[StructNameIndex],
+) -> PartialVMResult<(Vec<Type>, bool)> {
+    let mut tys = Vec::with_capacity(toks.len());
+    let mut all_fully_instantiated = true;
+    for tok in toks {
+        let (ty, is_fully_instantiated) = intern_type_impl(module, tok, struct_name_table)?;
+        tys.push(ty);
+        all_fully_instantiated &= is_fully_instantiated;
+    }
+    Ok((tys, all_fully_instantiated))
+}
+
+/// Returns the type, and true if this type is fully instantiated, and false otherwise.
+fn intern_type_impl(
```

Comment:
This is not really doing internalization, we shouldn't call it like this.

Specifically with the new module `ty_interner`.

Reply:
Changed to `convert_tok_to_type`, etc.